apps/src/lab2/views/components/IconButtonWithTooltip.tsx

```diff
@@ -11,14 +17,17 @@ interface IconButtonWithTooltipProps {
   icon: ButtonProps['icon'];
   type: ButtonProps['type'];
   color: ButtonProps['color'];
-  buttonSize: ButtonProps['size'];
+  buttonSize?: ButtonProps['size'];
   tooltipSize: TooltipProps['size'];
   tooltipDirection: TooltipProps['direction'];
-  hideTooltipTail: TooltipProps['hideTail'];
+  hideTooltipTail?: TooltipProps['hideTail'];
   disabled?: boolean;
-  onClick: () => void;
+  onClick?: () => void;
   containerRef?: React.RefObject<HTMLDivElement>;
   className?: string;
+  theme?: Theme;
+  href?: LinkButtonProps['href'];
+  target?: LinkButtonProps['target'];
```

Comment:
Ah bummer that we can't just use one set of props (ButtonProps) to handle both href and onClick - I thought this would be less complicated than it actually was 😄 thanks for doing this!